Clang 14 `unused-but-set` warnings in tests

With clang 14, I am encountering 3 `unused-but-set-variable` warnings for causing `Werror` to abort the build with a defualt Cmake configuration (`cmake ..`).

This is not covered by CI, as `ubuntu-compilers.yml` [`compiler-all-libs (clang-14, clang++-14, Release)`](https://github.com/Chaste/Chaste/actions/runs/8323708039/job/22773838603) does not encounter these warnings as it does not build the tests.

```
chaste/global/test/TestStaticPolymorphism.hpp:55:13: error: variable 'total' set but not used [-Werror,-Wunused-but-set-variable]
        int total = 0;
            ^
chaste/global/test/TestStaticPolymorphism.hpp:76:13: error: variable 'total' set but not used [-Werror,-Wunused-but-set-variable]
        int total = 0;
            ^
chaste/lung/test/ventilation/TestAcinarUnitModels.hpp:247:15: error: variable 'flow_integral' set but not used [-Werror,-Wunused-but-set-variable]
       double flow_integral = 0.0;
              ^
```
https://github.com/Chaste/Chaste/blob/f2ff7ee04e70ac9d06c57344df8d017dbb12b97b/global/test/TestStaticPolymorphism.hpp#L53-L60

https://github.com/Chaste/Chaste/blob/f2ff7ee04e70ac9d06c57344df8d017dbb12b97b/global/test/TestStaticPolymorphism.hpp#L74-L81

https://github.com/Chaste/Chaste/blob/f2ff7ee04e70ac9d06c57344df8d017dbb12b97b/lung/test/ventilation/TestAcinarUnitModels.hpp#L246-L274


Possible fixes:

+ Suppression:

    ```
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Wunused-but-set-variable"
    // ...
    #pragma clang diagnostic pop
    ```

+ Actually use the set variables within the tests
    + E.g. assert against them, return them from the method etc.
+ Remove them, if the test still has meaning without them.

--- a/global/test/TestStaticPolymorphism.hpp
+++ b/global/test/TestStaticPolymorphism.hpp
@@ -52,7 +52,7 @@ public:
 
     void Run()
     {
-        int total = 0;
+        [[maybe_unused]] int total = 0;
         for (int i=0; i<1e8; i++)
         {
             total += Method(i);
@@ -73,7 +73,7 @@ public:
 
     void Run()
     {
-        int total = 0;
+        [[maybe_unused]] int total = 0;
         for (int i=0; i<1e8; i++)
         {
             total += Method(i);

--- a/lung/test/ventilation/TestAcinarUnitModels.hpp
+++ b/lung/test/ventilation/TestAcinarUnitModels.hpp
@@ -244,7 +244,7 @@ public:
        double pleural_pressure = 0.0;
 
        double ode_volume = 0.0;
-       double flow_integral = 0.0;
+       [[maybe_unused]] double flow_integral = 0.0;
 
        while (!time_stepper.IsTimeAtEnd())
        {

